src/mpid/ch4/netmod/ofi/ofi_am_impl.h

```diff
@@ -122,6 +122,64 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_am_init_request(const void *am_hdr,
     return mpi_errno;
 }
 
+/* This function allocates a new OFI_am_send_request_t for the given MPIR_Request. The new OFI
+ * am request is PREPEND to the list of all am send requests. This provides an easy way to
+ * access the latest allocated am send request. There are two notes about this:
+ * 1. The parent MPIR_Request must have an correctly initialized  req_hdr from
+ *    MPIDI_OFI_am_init_request function above. This is because the req_hdr in the
+ *    OFI_am_send_request are expecting to refer to the am_hdr and am_hdr_sz in the req_hdr of
+ *    the parent request.
+ * 2. The new ofi am send request is prepend to the send_req list in MPIR_Request. In this way,
+ *    The following send operation will use this one. */
```

Comment:
The following functions just create and releases the "request" object, there is no enqueue or dequeue code here, so having those comments here is confusing. If necessary, move those comment where enqeue/dequeue happens.

Reply:
This is old description. will cleanup